optaplanner-examples/src/main/java/org/optaplanner/examples/common/experimental/impl/IntervalClusterImpl.java

```diff
@@ -217,4 +169,52 @@ public String toString() {
                 ", hasOverlap=" + hasOverlap +
                 '}';
     }
+
+    // TODO: Make this incremental by only checking between the interval's start and end points
```

Comment:
Maybe we should fix that?

Reply:
Easier said then done. An interval start point/end point may be under an interval, like so:

------------     --------------
        ------------

which make activeIntervals > 0 at the interval start point. We would either need to memoize the active interval count at all start points, or determine how many active intervals there are before/after it.